pkg/registry/registry.go

```diff
@@ -27,9 +27,9 @@ import (
 
 // GetRegistryRefreshTokenFromAADExchange retrieves an OAuth2 refresh token for the registry specified by serverURL
 func GetRegistryRefreshTokenFromAADExchange(serverURL string, principalToken *adal.ServicePrincipalToken, tenantID string) (string, error) {
-
-	err := principalToken.EnsureFresh()
-	if err != nil {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTimeOut)
+	defer cancel()
+	if err := principalToken.EnsureFreshWithContext(ctx); err != nil {
```

Comment:
That's an additional field on the `adal.ServicePrincipalToken` that we should set  `MaxMSIRefreshAttempts=1`